### ISSUE TITLE
Saving stage damage and opening for edit threw exception

### ISSUE THREAD

**Issue by codymccoy** [HydrologicEngineeringCenter/HEC-FDA#652]
I saved computed stage damage curves and then tried to open for edit and the following exception was thrown:
![image](https://user-images.githubusercontent.com/47757848/201988100-fffe128c-2e8c-4ec0-944e-479324a972c5.png)

I will work on this right now.

**Comment by rnugent3** [comment]
Aha. So we need an empty constructor on Histogram and ThreadsafeInlineHistogram.

**Comment by codymccoy** [comment]
Sure. You can do it.

**Comment by rnugent3** [comment]
hmmmm. we do have empty constructors in Histogram and in ThreadsafeInlineHIstogram

**Comment by codymccoy** [comment]
Ok, let me take a look.

**Comment by codymccoy** [comment]
Looks like it is histogram data provider
![image](https://user-images.githubusercontent.com/47757848/201996694-7b23c798-693e-4941-91ad-479b589c5a5b.png)

**Comment by codymccoy** [comment]
HistogramDataProvider doesn't have an empty ctor. @rnugent3 do you still want to make the change on this one?

**Comment by rnugent3** [comment]
I'll let you make the change - I imagine working in that class will be easier for you than for me.

**Comment by codymccoy** [comment]
![image](https://user-images.githubusercontent.com/47757848/202034248-e278a9e3-5844-433b-acaa-63fa691f3983.png)

Ran into another one. I don't like all this Activator stuff. I am guessing the histogram class use to live at Statistics.Distributions.Histogram. It doesn't live there anymore.

**Comment by codymccoy** [comment]
![image](https://user-images.githubusercontent.com/47757848/202034415-7edaa08d-db2c-4554-a2e9-7dc3ae28abd5.png)
Looks like it is in Statistics.Histograms.Histogram. I will try that and see what happens.

**Comment by codymccoy** [comment]
I was just going to change line 55 but i see that all the distributions seem to use this method. Can we move histogram back to distributions or I can write some custom logic.

**Comment by codymccoy** [comment]
After looking through the git history on that class, it looks like it has always been where it is. I'm guessing we just haven't tried this logic before because they stage damages weren't saving before today.

**Comment by codymccoy** [comment]
I think i will pass this off to @Brennan1994 I think Brennan understands this logic a little better than I do. 
I added the following logic hoping that it would solve the problem but it just kicked the can down the road a bit. I really hate that logic because it will break if the class name or the namespaces ever changes but I couldn't think of anything else to do.
![image](https://user-images.githubusercontent.com/47757848/202038303-311572e0-0513-450e-b0a3-833ba3bdb31a.png)

After the above change I get this exception:
![image](https://user-images.githubusercontent.com/47757848/202038341-0a221108-33b9-46a1-8869-cf7c2e5dced1.png)

The histogram data provider gets its data cleared and then it tries to evaluate the zeroth item. This makes me think that there must be special logic needed in the AddUnlinkedRow(i) call.
![image](https://user-images.githubusercontent.com/47757848/202038625-1cc7c459-ee00-40d2-aa7d-fb5f7a61811e.png)

When i go to that method it says this:
![image](https://user-images.githubusercontent.com/47757848/202038666-f1afb71c-cb5f-407f-b7bb-457d7cfe4632.png)

**Comment by codymccoy** [comment]
I am on branch stage_damage_saving.

**Comment by rnugent3** [comment]
> ![image](https://user-images.githubusercontent.com/47757848/202034248-e278a9e3-5844-433b-acaa-63fa691f3983.png)
> 
> Ran into another one. I don't like all this Activator stuff. I am guessing the histogram class use to live at Statistics.Distributions.Histogram. It doesn't live there anymore.

statistics.histograms now

**Comment by codymccoy** [comment]
@Brennan1994 It looks like your fix is working well. I ran into a null pointer and needed to add a name to the empty ctor in the data provider.
![image](https://user-images.githubusercontent.com/47757848/202777142-30ffee54-be41-4781-89d0-3c088e483779.png)
I am now able to open up the stage damages in edit mode. The values in the table with plot are not correct when opening up for edit, but I think that is my problem and I will make that another issue.

This is completed in branch stage_damage_saving.

**Comment by rnugent3** [comment]
This work is finished and ready to be tested.

**Comment by rnugent3** [comment]
We can return to this issue after #686 is resolved

**Comment by rnugent3** [comment]
this issue has been resolved
